workers/unity/Packages/com.improbable.gdk.querybasedinterest/InterestComponent.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Improbable.Worker.CInterop;
+using UnityEngine;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    public class InterestComponent
+    {
+        private Dictionary<uint, ComponentInterest> interest;
+
+        public InterestComponent()
+        {
+            interest = new Dictionary<uint, ComponentInterest>();
+        }
+
+        public InterestComponent AddComponentInterest<T>(ComponentInterest componentInterest)
+            where T : ISpatialComponentData
+        {
+            var componentId = Dynamic.GetComponentId<T>();
+            if (interest.ContainsKey(componentId))
+            {
+                Debug.LogWarning($"Skipping component with id {componentId}.");
+            }
+            else
+            {
+                interest.Add(componentId, componentInterest);
+            }
+            
+            return this;
+        }
+
+        public InterestComponent AddQuery<T>(ComponentInterest.Query interestQuery)
+            where T : ISpatialComponentData
+        {
+            var componentId = Dynamic.GetComponentId<T>();
+            if (!interest.TryGetValue(componentId, out var query))
+            {
+                return AddComponentInterest<T>(new ComponentInterest(new List<ComponentInterest.Query>
```

Comment:
What happens in this branch? Does it add a component interest for the query? Is it a shortcut? Why?